pkg/manifest/firstboot.go

```diff
@@ -0,0 +1,168 @@
+package manifest
+
+import (
+	"fmt"
+	"html/template"
+	"io/fs"
+	"strings"
+
+	"github.com/osbuild/images/internal/common"
+	"github.com/osbuild/images/pkg/customizations/firstboot"
+	"github.com/osbuild/images/pkg/customizations/fsnode"
+	"github.com/osbuild/images/pkg/osbuild"
+)
+
+// checkName prevents path traversal
```

Comment:
While this is certainly a side-effect of this function it's also overly restrictive for preventing path traversal (`systemd-firstboot-первый-старт.service` is a perfectly valid filename). This seems better suited to live in the blueprint library?

In `images` we'd then take the `basename` (to be defensive and have some locality) and then throw it through `systemd-escape` when we put it in the stage (since `systemd` has its own ideas for escaping). We already have a helper for the last bit in `internal/common/helpers.go` (`MountUnitNameFor`) that could be duplicated (or expanded) to also handle service names?

Reply:
I think using `systemd-escape` is an overkill, but you are right that there is no reason to be too restrictive. I will just use `filepath.Base(userInput)` to do the same job.